upstairs/src/lib.rs

```diff
@@ -4299,6 +4776,10 @@ async fn process_new_io(
             *lastcast += 1;
         }
         BlockOp::Flush => {
+            if !up.guest_io_ready() {
```

Comment:
why is this check only here, and not under Read and Write too?

Reply:
For read and write, we check in the `submit_` method.  For the flush, since the upstairs will call submit_flush after we have set deactivate, we can't do the check there because it will block our internally generated flush.

Would it make better sense to put the check here for all three, to at least be consistent?

Reply:
I'd go with a comment here instead :)

Reply:
Comment added in a0e30e3